src/App.tsx

```diff
@@ -1,25 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const handlePerPageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(Number(e.target.value));
+    setCurrentPage(1);
+  };
+
+  const startIndex = (currentPage - 1) * perPage;
+  const endIndex = startIndex + perPage;
```

Comment:
The calculation of `startIndex` and `endIndex` could be moved inside a `useMemo` hook to optimize performance. This would prevent these values from being recalculated on every render when the current page and items per page have not changed.